### PR TITLE
Remove install dependency + support Python 3.9, 3.10

### PR DESCRIPTION
**Pull request by minouHub**
This PR is to attempt to remove dependencies in the install of RADIS. Each push will be tested for one of the requirements of `environment.yaml`

- [x] python=3.8  - required by vaex, see https://github.com/vaexio/vaex/issues/2397
- [x] cython
- [x] pandas < 2.0 - solved in #641
- [x] publib - Not possible with python>3.8
- [x] progressbar2 - noted as needed by vaex, so it should be automatically managed by vaex (not us)

Also:

- [x] add limit `python<3.9` in `setup.py` for pip install

### CODE REVIEW

File: environment.yml
Comment (erwanp) [inline environment.yml:22]:
```suggestion
- pandas
```

Comment (erwanp) [reply environment.yml:22]: I'd keep pandas itself, but release the version constraint

**Review by erwanp — CHANGES_REQUESTED** [review]

**Comment by erwanp** [comment]
Also upgrade publib dependency in requirements.txt to 
```
publib>=0.4.0
```

I just published a new version of the package (https://pypi.org/project/publib/ ) that fixes a non compatibility with Matplotlib 3.8.0  
due to https://github.com/matplotlib/matplotlib/issues/26831

**Comment by erwanp** [comment]
Regarding the test failures, see my comment here:  https://github.com/radis/radis/pull/633/files#r1555134045

File: requirements.txt
Comment (erwanp) [inline requirements.txt:4]: therefore test on python>=3.11 or something?

Comment (minouHub) [reply requirements.txt:4]: sorry I didn't get it.

Comment (minouHub) [reply requirements.txt:4]: vaex is not compatible with python>3.8 (see original post). matplotlib==3.8 requires python>3.8, thus, we should hold this update for the moment

**Comment by minouHub** [comment]
@erwanp, waiting for your final review on this. I will try to sort the docs by tonight

**Comment by erwanp** [comment]
Which is actively being worked : https://github.com/vaexio/vaex/pull/2331

**Comment by minouHub** [comment]
@erwanp did you change this PR name because you want to wait for vaex? I would have prefer to merge now and put the support of Python 3.11 on a to-do list

**Comment by erwanp** [comment]
@minouHub  ok I updated the PR title

**Comment by HajimeKawahara** [comment]
@minouHub @erwanp 

We have an installation problem because of 
- add limit python<3.9 in setup.py for pip install

What is the reason for this restriction?

https://github.com/HajimeKawahara/exojax/pull/481

**Comment by minouHub** [comment]
Hello,
This restriction is introduced because `vaex` cannot be installed with Python > 3.8. Before this PR, the limit was present in the `conda` requirements but not in `pip`. In other words, if you did that, the **regular** pip install failed, see below:
```
conda create --name test_radis python==3.9 # or anything above 3.9
conda activate test_radis
pip install radis
```
The problem was that the error message was hard to understand. With this requirement, it's clearer. In the past, you could somehow trick the install with the developer install `pip -e .` but this is not stable to my opinion.

The main strategy I applied was to wait for `vaex` to update their package. The alternative if that is a big problem for you is to revert this modification, but remains the installation problem.

**Comment by HajimeKawahara** [comment]
@minouHub Thanks. I see. The problem is that jax requires python >3.8. so, for us, it is a big problem. Can you revert this modification for a while if you do not have a big problem?

**Comment by minouHub** [comment]
oh wait .... it worked .... I'm trying a quick fix in #651. Will keep you updated

**Comment by HajimeKawahara** [comment]
Ah, I see. Thanks for the quick check!

**Comment by minouHub** [comment]
Hello,
Very sorry for the mess. Two mistakes on my side. 

1. `vaex` is actually compatible with python 3.9 and 3.10. I don't understand how I missed that.
2. if `vaex` limits the python version, we don't need to state it on our side

**Comment by HajimeKawahara** [comment]
@minouHub thank you so much!

**Comment by erwanp** [comment]
> 2. if `vaex` limits the python version, we don't need to state it on our side

If and only if vaex is installed through Conda and not Pip on our side; which it is=> so all good. Well done for fixing it

File: requirements.txt
Comment (minouHub) [reply requirements.txt:4]: note: vaex is compatible up to python 3.10

**Pull request merged**
